sys-devel/distcc/distcc-3.4-r7.ebuild

```diff
@@ -0,0 +1,212 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..13} )
+
+inherit autotools flag-o-matic prefix python-single-r1 systemd xdg
+
+DESCRIPTION="Distribute compilation of C code across several machines on a network"
+HOMEPAGE="https://github.com/distcc/distcc"
+SRC_URI="https://github.com/distcc/distcc/releases/download/v${PV}/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~m68k ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86"
+IUSE="gssapi gtk hardened ipv6 selinux xinetd zeroconf"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+RDEPEND="
+	${PYTHON_DEPS}
+	dev-libs/popt
+	gssapi? ( net-libs/libgssglue )
+	gtk? ( x11-libs/gtk+:3 )
+	zeroconf? ( >=net-dns/avahi-0.6[dbus] )
+"
+DEPEND="
+	${RDEPEND}
+	sys-libs/binutils-libs
+"
+BDEPEND="
+	${PYTHON_DEPS}
+	dev-build/autoconf-archive
+	virtual/pkgconfig
+"
+RDEPEND+="
+	acct-user/distcc
+	>=dev-util/shadowman-4
+	>=sys-devel/gcc-config-1.4.1
+	selinux? ( sec-policy/selinux-distcc )
+	xinetd? ( sys-apps/xinetd )
+"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-3.0-xinetd.patch"
+	# SOCKSv5 support needed for Portage, bug #537616
+	"${FILESDIR}/${PN}-3.2_rc1-socks5.patch"
+	"${FILESDIR}/${P}-fix-dcc_gcc_rewrite_fqn-corruption.patch"
+	"${FILESDIR}/${P}-rewrite-chost.patch"
+)
+
+src_prepare() {
+	default
+
+	# Bugs #120001, #167844 and probably more. See patch for description.
+	use hardened && eapply "${FILESDIR}/distcc-hardened.patch"
+
+	sed \
+		-e "s:@EPREFIX@:${EPREFIX:-/}:" \
+		"${FILESDIR}/distcc-config-r1" > "${T}/distcc-config" || die
+
+	# TODO: gdb tests fail due to gdb failing to find .c file
+	sed -i -e '/Gdb.*Case,/d' test/testdistcc.py || die
+
+	# see rename comment in src_install
+	if use gtk; then
+		sed -i -e 's/distccmon-gnome/distccmon-gui/g' gnome/distccmon-gnome.desktop || die
+	fi
+
+	hprefixify update-distcc-symlinks.py src/{serve,daemon}.c
+	python_fix_shebang update-distcc-symlinks.py "${T}/distcc-config"
+	eautoreconf
+}
+
+src_configure() {
+	# https://github.com/distcc/distcc/issues/454
+	append-cppflags -DPY_SSIZE_T_CLEAN
```

Comment:
note to myself

try and backport https://github.com/distcc/distcc/pull/479 to re-enable pumpmode